app/views/RoomView/index.tsx

```diff
@@ -1345,6 +1345,7 @@ class RoomView extends React.Component<IRoomViewProps, IRoomViewState> {
 			content = (
 				<Message
 					item={item}
+					index={index}
```

Comment:
We shouldn't rely on index to do this.
It's based on the message position inside of the [array of messages](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/7934141d31dab125c4c4ed75edbc2b0e1e9c40fc/app/views/RoomView/List/index.tsx#L25), which is going to change [depending on the usage or new messages](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/7934141d31dab125c4c4ed75edbc2b0e1e9c40fc/app/views/RoomView/List/hooks/useMessages.ts#L83).

A much more secure solution would be to query local database for next messages inside the room by looking at messages greater than this [message's timestamp](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/7934141d31dab125c4c4ed75edbc2b0e1e9c40fc/app/lib/database/model/Message.js#L19)

Reply:
Sure, thank you, sir, for reviewing my PR. I'll make the necessary changes.